webapps/frontend/app/viewmodels/wizards/addvdisk/gather.js

```diff
@@ -0,0 +1,131 @@
+// Copyright 2014 iNuron NV
+//
+// Licensed under the Open vStorage Modified Apache License (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.openvstorage.org/license
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/*global define */
+define([
+    'jquery', 'knockout',
+    'ovs/shared', 'ovs/api', 'ovs/generic', './data', '../../containers/vpool', '../../containers/storagerouter'
+], function($, ko, shared, api, generic, data, VPool, StorageRouter) {
+    "use strict";
+    return function() {
+        var self = this;
+
+        // Variables
+        self.data   = data;
+        self.loadStorageRoutersHandle = undefined;
+        self.loadvPoolsHandle = undefined;
+
+        // Observables
+
+        // Computed
+        self.canContinue = ko.computed(function() {
+            var valid = true, showErrors = false, reasons = [], fields = [];
+            if (self.data.name() === undefined || self.data.name() === '' || !self.data.name.valid()) {
+                valid = false;
+                fields.push('name');
+                reasons.push($.t('ovs:wizards.add_vdisk.gather.invalid_name'));
+            };
+            return { value: valid, showErrors: showErrors, reasons: reasons, fields: fields };
+        });
+
+        self.activate = function() {
+            generic.xhrAbort(self.loadStorageRoutersHandle);
+            self.loadStorageRoutersHandle = api.get('storagerouters', {
+                queryparams: {
+                    contents: 'storagedrivers',
+                    sort: 'name'
+                }
+            })
+                .done(function(data) {
+                    var guids = [], srdata = {};
+                    $.each(data.data, function(index, item) {
+                        guids.push(item.guid);
+                        srdata[item.guid] = item;
+                    });
+                    generic.crossFiller(
+                        guids, self.data.storageRouters,
+                        function(guid) {
```

Comment:
You _must_ return an object in this function. Not returning anything will cause strange behavior (or undefined objects - which will make other code more complex to handle this). If there's a guid which you don't want included, you need to filter these out before adding them to the ```crossFiller```. So basically move the below if-statement to ```$.each(...)``` above.